### ISSUE TITLE
bug : in ollama llm the version is not so in drop down 

### ISSUE THREAD

**Issue by itsKayumkhan** [stackblitz-labs/bolt.diy#869]
### Describe the bug

![image](https://github.com/user-attachments/assets/c640928a-abf4-424d-aeea-8c2054e795ea)

the version is not show in 7b version of ollama

### Link to the Bolt URL that caused the error

http://localhost:3000/chat/3

### Steps to reproduce

go to home page 
select the ollama model

### Expected behavior

not show

### Screen Recording / Screenshot

![image](https://github.com/user-attachments/assets/fa360b7f-fa2d-4acb-86c0-99084dcc13ae)

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

### Provider Used

brave

### Model Used

ollama

**Comment by Soumyaranjan-17** [comment]
Are you running ollama
check it on http://127.0.0.1:11434
or on http://[127.0.0.1:11434/api/tags](http://127.0.0.1:11434/api/tags)

**Comment by afhanxd** [comment]
I am facing the same issue, the model does not appear there, i followed all steps, ollama is running checked from http://127.0.0.1:11434/ but still same issue

**Comment by Soumyaranjan-17** [comment]
Can you show some screenshots

**Comment by afhanxd** [comment]
Here are some screenshots 

![Screenshot 2024-12-23 092001](https://github.com/user-attachments/assets/01fae74b-44d2-4a47-bbc8-c95658d9b1a2)
![Screenshot 2024-12-23 092011](https://github.com/user-attachments/assets/6856132e-57dd-4c55-983c-6af5a1844f86)
![Screenshot 2024-12-23 092103](https://github.com/user-attachments/assets/16c510c2-2fb8-498c-bde2-14f90f64486d)
![Screenshot 2024-12-23 092126](https://github.com/user-attachments/assets/22c704d7-716e-4951-9256-7bfa755244d6)

**Comment by amanguleria-1** [comment]
You will have to go to the providers from the settings tab and scroll down to ollama and paste ollama base url as - [  http://localhost:11434  ]   (default port used to interact with ollama). Refresh and it shows up now.

**Comment by afhanxd** [comment]
Damn Bro, it worked. thank u for helping me out but im having another problem now it shows this 

![image](https://github.com/user-attachments/assets/75254aec-caff-4c12-b2bf-4b1fdc597595)

**Comment by Laslink** [comment]
me too whit ollama errno: -4078,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '::1',
port: 11434

**Comment by Soumyaranjan-17** [comment]
Use http://127.0.0.1:11434 instead of localhost: 11434
In .env and in BOLT UI setting provider also

**Comment by Soumyaranjan-17** [comment]
> me too whit ollama or api key openia errno: -4078, code: 'ECONNREFUSED', syscall: 'connect', address: '::1', port: 11434

Don't use http://localhost:11434
Use http://127.0.0.1:11434 

Because IPV6 conversation will convert the localhost to ::1:11434 which will make some issue

**Comment by afhanxd** [comment]
@Soumyaranjan-17 bruhhh it worked 😭😭😭 thanks. Also i am only able to use 05b-1.5b anything above that just makes my laptop lag.

Laptop

16GB LPDDR5X
NVIDIA® GeForce RTX™ 4060 Laptop GPU (233 AI TOPs)
8GB GDDR6
AMD Radeon™ Graphics 
AMD Ryzen™ 9 7940HS Mobile Processor 4.0GHz (8-core/16-thread, 16MB cache, up to 5.2GHz max boost)

**Comment by Soumyaranjan-17** [comment]
@afhanxd Congratulations bro....
But actually I don't have knowledge about the system hardware 
Because I only have 6GB of ram without any dedicated graphics card 😐

But I think qwen 7b must work on your computer. As it works in mine also 😅

If any issues you are facing show some screenshots of errors and config.

**Comment by Mayur1804** [comment]
Can someone help me with this why I'm not able to access

<img width="1768" alt="image" src="https://github.com/user-attachments/assets/7865183e-f385-4605-9f99-d896d7832009" />
<img width="1628" alt="image" src="https://github.com/user-attachments/assets/28dc701e-a8af-447e-952a-25f410de378e" />

**Comment by oliverbarreto** [comment]
The same here... can you provide help !!!

Current Setup with Windows 11 Host:
- Native Ollama instance working perfectly fine with Open WebUI and python scripts
- Bolt.diy using Docker... and working PERFECTLY using Google API KEY and Gemini 2 Model... but not with Ollama local models
- Environment Variable for OLLAMA: OLLAMA_HOST=0.0.0.0:11434
- If I configure Ollama Provider in Bolt.diy as "http:127.0.0.1:11434"  nor "http://host_IP:11434" i cannot see available models in my working ollama server.
- If i configure Ollama Provider in Bolt.diy as "https://ollama.mydomain.duckdns.org/" and configure a reverse proxy host for ollama server with a DNS rule pointing to "host_IP:11434"... i can see the dropdown of available models, but when i hit send... I get the same error as previous comments

![Captura de pantalla 2025-01-01 182504](https://github.com/user-attachments/assets/cce6ae1a-09af-490d-b123-fb9b0de50a91)
![Captura de pantalla 2025-01-01 181307](https://github.com/user-attachments/assets/1f3f4e14-1581-4892-b49b-75cca9f641e3)
![Captura de pantalla 2025-01-01 181322](https://github.com/user-attachments/assets/dd23ea7e-a971-43cf-ac0c-36a01738ef44)

**Comment by Mayur1804** [comment]
> @Mayur1804 Have you installed the model that you are using !?

Yes. You can see in one of my screenshot saying Ollama is running

**Comment by nwslnk** [comment]
ymmv but hope this might help someone landing here:

worth noting that if you're running ollama or bolt via https, that both services should to be using the same scheme or the request to list models may fail in your browser (check console for mixed http/https errors)

ollama & bolt via traefik w/o origins
![image](https://github.com/user-attachments/assets/3565edc8-3014-4d6b-9ac6-4bc9899d6f27)

if youre running bolt with ssl and ollama in another container you need to enable ssl for ollama as well or you'll see CORS issues in the browser console.  You also need to pass something like -e OLLAMA_ORIGINS="*" (not in prod obviously)  (https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-allow-additional-web-origins-to-access-ollama)

ollama via http, bolt via https
![image](https://github.com/user-attachments/assets/51424941-e9c1-462a-9ae8-b75dc277ed08)

with bolt and ollama running over ssl via traefik with the origins env flag mentioned the model list loads correctly 

with origins & both via ssl
![image](https://github.com/user-attachments/assets/f38cbef6-1790-4e34-a2c3-c06b0bf24dee)


also fwiw... if you're using podman you may need to update the compose file from "docker" to "containers", and it may need to be run as root vs rootless

```
    extra_hosts:
      - "host.containers.internal:host-gateway"
```

**Comment by zha0jf** [comment]
@oliverbarreto 
I encountered this issue as well. I looked at the code in `app/lib/modules/llm/providers/ollama.ts` and found the following lines:
```typescript
// Backend: Check if we're running in Docker
const isDocker = process.env.RUNNING_IN_DOCKER === 'true';

baseUrl = isDocker ? baseUrl.replace('localhost', 'host.docker.internal') : baseUrl;
baseUrl = isDocker ? baseUrl.replace('127.0.0.1', 'host.docker.internal') : baseUrl;
```
If Bolt.diy is running in Docker and using Ollama, the OLLAMA_API_BASE_URL environment variable must be set. Otherwise, when the code reaches this point, it will throw an error and exit because the baseUrl is empty.
 After configuring this environment variable in the docker-compose.yaml, the issue was resolved。
